### PR TITLE
Cleanup, separate backup bootstrap and mgmt

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Refactor crypto test, separate 4S and backup mgmt test.

No Change in behavior.
Just split a describe in a wrapping describe with two describes, and moved around common functions.

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/integ/crypto/crypto.spec.ts [hunk 1]
```diff
@@ -2292,44 +2292,82 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("crypto (%s)", (backend: string,
...
+        function sendSyncResponse(secretStorageKey: string) {
+            syncResponder.sendOrQueueSyncResponse({
+                next_batch: 1,
+                account_data: {
+                    events: [
+                        {
+                            type: "m.secret_storage.default_key",
+                            content: {
+                                key: secretStorageKey,
+                                algorithm: SECRET_STORAGE_ALGORITHM_V1_AES,
```
Comment (weeman1337) [inline spec/integ/crypto/crypto.spec.ts:2336]: The `default_key` does not have the `algorithm` field according [to the spec](https://spec.matrix.org/v1.8/client-server-api/#key-storage). Should this removed here to test nearer to reality?

File: spec/integ/crypto/crypto.spec.ts [hunk 2]
```diff
@@ -2292,44 +2292,82 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("crypto (%s)", (backend: string,
...
+                            },
+                        },
+                        // Needed for secretStorage.getKey or secretStorage.hasKey
+                        {
+                            type: `m.secret_storage.key.${secretStorageKey}`,
+                            content: {
+                                key: secretStorageKey,
```
Comment (weeman1337) [inline spec/integ/crypto/crypto.spec.ts:2343]: The `key` property doesn't exist [according to the spec](https://spec.matrix.org/v1.8/client-server-api/#msecret_storagev1aes-hmac-sha2). Should this removed here to test nearer to reality?

**Review by weeman1337 — APPROVED** [review]
Looks good :+1:

Two comments in moved code about tests being spec compliant. It was already like this before. But maybe we can improve here.

File: spec/integ/crypto/crypto.spec.ts (hunk 1 above)
Comment (BillCarsonFr) [reply spec/integ/crypto/crypto.spec.ts:2336]: good catch!

File: spec/integ/crypto/crypto.spec.ts (hunk 2 above)
Comment (BillCarsonFr) [reply spec/integ/crypto/crypto.spec.ts:2343]: good catch!

**Pull request merged**